https://github.com/protocolbuffers/protobuf/commit/decc1b570880709a84e3c7200e637fee832098c5.patch

From decc1b570880709a84e3c7200e637fee832098c5 Mon Sep 17 00:00:00 2001
From: Karthikeyan Singaravelan <tir.karthi@gmail.com>
Date: Mon, 20 Mar 2023 23:16:01 -0700
Subject: [PATCH] Update deprecated aliases for Python 3.12 compatibility.
 (#12254)

https://docs.python.org/3.12/whatsnew/3.12.html#removed . `assertRaisesRegexp` was removed in favor of `assertRaisesRegex`

Closes #12254

COPYBARA_INTEGRATE_REVIEW=https://github.com/protocolbuffers/protobuf/pull/12254 from tirkarthi:fix-assert 922f3ceccc607f0c8b56519fd13e01d20c04909c
PiperOrigin-RevId: 518181080
---
 python/google/protobuf/internal/json_format_test.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git google/protobuf/internal/json_format_test.py google/protobuf/internal/json_format_test.py
index 893451a03cb0f..b4f516d834409 100644
--- google/protobuf/internal/json_format_test.py
+++ google/protobuf/internal/json_format_test.py
@@ -1060,7 +1060,7 @@ def testInvalidMap(self):
   def testInvalidTimestamp(self):
     message = json_format_proto3_pb2.TestTimestamp()
     text = '{"value": "10000-01-01T00:00:00.00Z"}'
-    self.assertRaisesRegexp(
+    self.assertRaisesRegex(
         json_format.ParseError, 'Failed to parse value field: '
         'time data \'10000-01-01T00:00:00\' does not match'
         ' format \'%Y-%m-%dT%H:%M:%S\' at TestTimestamp.value.',
@@ -1097,7 +1097,7 @@ def testInvalidTimestamp(self):
   def testInvalidOneof(self):
     message = json_format_proto3_pb2.TestOneof()
     text = '{"oneofInt32Value": 1, "oneofStringValue": "2"}'
-    self.assertRaisesRegexp(
+    self.assertRaisesRegex(
         json_format.ParseError, 'Message type "proto3.TestOneof"'
         ' should not have multiple "oneof_value" oneof fields at "TestOneof".',
         json_format.Parse, text, message)
